### DISCUSSION TITLE
App Basics : Newbie haha pls help M1 MacBook Air

### DISCUSSION THREAD

**Discussion by eap17ande** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#231] [Q&A]
Hello! 
I am possibly one of the least tech-savvy people alive so please bear with me. 
Basically I plan on keeping my laptop plugged in (for many days) and I wanted to see if there was a way to like save as much of the battery as possible. I thus stumbled across this app (thank u reddit) and downloaded it. 
I was just wondering if someone could help me understand what's going on because I looked through the discussions and noticed that my thing didn't look like the other things so I'm wondering if I did something wrong.

I am using a M1 MacBook Air running on macOS Big Sur Version 11.1

So this is what it currently looks like (it is plugged in, I already reset my device after downloading)  : 
<img width="407" alt="Screen Shot 2021-04-03 at 1 06 32 PM" src="https://user-images.githubusercontent.com/81879887/113490202-81891680-947d-11eb-8f04-4b1890ad3867.png">
<img width="338" alt="Screen Shot 2021-04-03 at 1 06 40 PM" src="https://user-images.githubusercontent.com/81879887/113490206-82ba4380-947d-11eb-9344-d85e976b3674.png">

And I was wondering the following : 
1. What is Charge Inhibit (what does it mean for it to be True or False)?
     a) If it is False how do I make it True and vice versa. 
     b) To Optimize battery health/longevity what should I have it as?
2. What is Prevent sleep (what does it mean for it to be True or False)?
     a) If it is False how do I make it True and vice versa. 
     b) To Optimize battery health/longevity what should I have it as?
3. What is Prevent sleep (what does it mean for it to be True or False)?
     a) If it is False how do I make it True and vice versa. 
     b) To Optimize battery health/longevity what should I have it as?
4. What is Launch at login (what does it mean for it to be checked or unchecked)?
     a) To Optimize battery health/longevity what should I have it as?
5. What is Allow discharge (what does it mean for it to be checked or unchecked)?
     a) To Optimize battery health/longevity what should I have it as?
And finally, does everything look right? I don't know why I feel like it's not working (well part of why I feel like it's not working is because when I click on battery it doesn't show that it's plugged in and not using battery)

To whoever takes the time to read and answer this, I deeply thank you for your time, patience, and response

**Comment by danfoerster** [comment]
I can't specifically tell you what each of these things mean (I am on an Intel Macbook with the older Mode enabled and it works a little different there) but I am guessing
Charge Inhibit: Wether AlDente is actively stopping current reaching the battery, meaning, if 'False', it is not inhibiting charging and if 'True' AlDente would be in full effect.
Prevent Sleep: I think this is mostly 'True' as the Allow Discharge makes it run in the background and attempting to drain the battery to any set value whenever possible, which could affect how the device goes into sleep mode.
Inflow Enabled: Wether the machine receives any power. Not necessarily charging the battery but by powering the machine directly.

However I doubt that, since you don't really have any control over wether those things are True or False, that they affect battery health at all.
You also have to leave AlDente in the menu bar and not quit it! (on M1 Macs)

I'd try disabling 'Allow Discharge', because as for now, your Mac is going to try to drain the battery until it reaches the 80% you set in AlDente. It's basically forcing the use of the battery and then switches to power from the power adapter directly. It should look like this (picture) and it should say 'Power Source: Power Adapter' and 'Battery not charging' (right now for you it says 'Power Source: Battery' still because AlDente will be 'bypassed' until it is drained to the 80%). Once 80% is reached, the battery icon should switch to the prong thingy.

<img width="313" alt="Screen Shot 2021-04-05 at 11 34 04" src="https://user-images.githubusercontent.com/79795921/113560156-d6a26500-9602-11eb-955a-6940e6b30712.png">

If you have your Mac plugged in for longer, set the value to roughly 60/70%ish. Wait til the charging stops (menu bar icon changes), then change the value to something lower. Make sure 'Allow Discharge' is disabled for this. Your Mac will then not recharge until the lower percentage is reached. Then you will have to set AlDente to your max value again and it will continue charging. Once charging stops, put in the lower value again. (Once every other month, charge completely to 100% and then quit Aldente, so the battery can dicharge til empty. Then charge again fully. Reason for doing that is to calibrate the battery meter again, since it can become quite inaccurate if the battery is always left on a partial charge level etc.

Hope this helps you a little bit :)

**Reply by Dhudong** [reply]
> I can't specifically tell you what each of these things mean (I am on an Intel Macbook with the older Mode enabled and it works a little different there) but I am guessing
> Charge Inhibit: Wether AlDente is actively stopping current reaching the battery, meaning, if 'False', it is not inhibiting charging and if 'True' AlDente would be in full effect.
> Prevent Sleep: I think this is mostly 'True' as the Allow Discharge makes it run in the background and attempting to drain the battery to any set value whenever possible, which could affect how the device goes into sleep mode.
> Inflow Enabled: Wether the machine receives any power. Not necessarily charging the battery but by powering the machine directly.
> 
> However I doubt that, since you don't really have any control over wether those things are True or False, that they affect battery health at all.
> You also have to leave AlDente in the menu bar and not quit it! (on M1 Macs)
> 
> I'd try disabling 'Allow Discharge', because as for now, your Mac is going to try to drain the battery until it reaches the 80% you set in AlDente. It's basically forcing the use of the battery and then switches to power from the power adapter directly. It should look like this (picture) and it should say 'Power Source: Power Adapter' and 'Battery not charging' (right now for you it says 'Power Source: Battery' still because AlDente will be 'bypassed' until it is drained to the 80%). Once 80% is reached, the battery icon should switch to the prong thingy.
> 
> <img alt="Screen Shot 2021-04-05 at 11 34 04" width="313" src="https://user-images.githubusercontent.com/79795921/113560156-d6a26500-9602-11eb-955a-6940e6b30712.png">
> 
> If you have your Mac plugged in for longer, set the value to roughly 60/70%ish. Wait til the charging stops (menu bar icon changes), then change the value to something lower. Make sure 'Allow Discharge' is disabled for this. Your Mac will then not recharge until the lower percentage is reached. Then you will have to set AlDente to your max value again and it will continue charging. Once charging stops, put in the lower value again. (Once every other month, charge completely to 100% and then quit Aldente, so the battery can dicharge til empty. Then charge again fully. Reason for doing that is to calibrate the battery meter again, since it can become quite inaccurate if the battery is always left on a partial charge level etc.
> 
> Hope this helps you a little bit :)

I think this covers everything there is to be discussed. Just one more thing, since you're on M1 mac, should your mac still charge up to 100% after setting your desired maximum charge (make sure you're using the version 2.2 of the app), I'd recommend that you try running it via Rosetta 2. You can do this by going to app library, right click on Al Dente and click on "Get info". After that, click on "Open using Rosetta". It should run as intended. 
<img width="268" alt="image" src="https://user-images.githubusercontent.com/82040077/113803594-6b39ce00-978f-11eb-81f3-547420df28ce.png">
